src/crypto/tests/CHIPCryptoPALTest.cpp

```diff
@@ -662,7 +664,12 @@ static void TestECDSA_Signing_SHA256_Msg(nlTestSuite * inSuite, void * inContext
     const char * msg  = "Hello World!";
     size_t msg_length = strlen(msg);
 
+#ifdef ENABLE_HSM_EC_KEY
+    P256KeypairHSM keypair;
```

Comment:
I think readability could be improved if you create a derived clas that is ifdefed instead of having `#ifdef` in so many places.

Reply:
+1, I think this needs some more design thought

Reply:
Using a separate derived class in the tests now. Parent class is decided based on #ifdef.